dataframe/dataframe.go

```diff
@@ -0,0 +1,45 @@
+package dataframe
+
+// DataFrame is an immutable data frame that support filtering, aggregation and
+// data manipulation.
+type DataFrame interface {
+	// Column returns a column identified by name, panics if not present.
+	Column(name string) Column
+	// Columns returns all columns present in the dataframe.
+	Columns() Columns
+
+	// Distinct returns a new DataFrame that only contains unique rows with
+	// respect to the specified columns. If no columns are given Distinct will
+	// return rows where allow columns are unique.
+	Distinct(columns ...Column) DataFrame
+
+	// Filter returns a new filtered DataFrame according to the filter.
+	Filter(filter Filter) DataFrame
+
+	// GroupBy groups rows together for which the values of specified columns
+	// are the same.
+	GroupBy(columns ...Column) Groups
```

Comment:
It would be good to see an example as `GroupBy` on multiple columns can be understood in many ways - does ordering matter for example? Or is it random?

Reply:
GroupBy ordering does not matter (no rows dissapear so not sure how ordering could matter?). I agree examples would be a good thing, documentation will be in next release.